Automattic/jetpack
# Update my jetpack zendesk chat show conditionally with api

## Proposed changes:

Add a REST API call to My Jetpack that queries Zendesk chat availability. This is then used to conditionally render the Zendesk chat widget

This uses the `/wpcom/v2/presales/chat?group=jp_presales` endpoint to determine whether or not to show the component

This, in turn, renders the conditional rendering inside the zendesk chat widget component itself pointless and is removed in this PR so it can be handled entirely by the endpoint

### Other information:


## Jetpack product discussion

P2: pbNhbs-5cD-p2#comment-14447

## Does this pull request change what data or activity we track or use?

No

## Testing instructions:

1. Check out this branch
2. Start your local environment or use the Jetpack Beta Plugin to use this branch
3. If you are using your local environment, run `jetpack install packages/my-jetpack && jetpack build packages/my-jetpack && jetpack watch packages/my-jetpack` to make sure you can follow the testing instructions easily
4. Connect Jetpack to your site if you haven't already
5. Go to https://developer.wordpress.com/docs/api/console/ and test out the `wpcom/v2` api with the endpoint `presales/chat?group=jp_presales`. Take note of what is returned. If `is_available` is `false`, you should not see the chat widget in the next step, if it is `true`, you should see it
6. Go to `/wp-admin/admin.php?page=my-jetpack` make sure you see/don't see the chat widget based on what happened in step 5. If `is_available` was `true`, you should see this
![Screenshot 2023-04-14 at 2 19 30 PM](https://user-images.githubusercontent.com/65001528/232147149-97c82a9f-b299-4683-a58b-1c01907f95ee.jpg)
7. Now, go to the file `projects/packages/my-jetpack/_inc/state/resolvers.js` and update line 75  to this if you were not able to see the chat widget the first time:
```js
{ is_available: true }
```
if you WERE able to see it the first time, change that line to this
```js
{ is_available: false }
```
If you saw it the first time, you should now no longer see it, and if you did not see it the first time, you should now see it.
